src/kagglehub/handle.py

```diff
@@ -47,7 +49,7 @@ def to_url(self) -> str:
 
 
 @dataclass
-class DatasetHandle:
+class DatasetHandle(ResourceHandle):
     owner: str
     dataset_name: str
```

Comment:
tiny nit - I know this isn't added code, but since it's early, can we please change this to `dataset` vs `dataset_name`.  A couple reasons:
* Name is somewhat redundant / doesn't add information
* Consistency with models
* `_name` usually refers to something human readable like "My Cool Dataset", but this is actually a slug like "my-cool-dataset".